bundles/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/internal/AutelisBinding.java

```diff
@@ -0,0 +1,353 @@
+		}
+		
+		int port = DEFAULT_PORT;
+		String host = (String) configuration.get("host");
+		String username = (String) configuration.get("username");
+		String password = (String) configuration.get("password");
+		
+		if(StringUtils.isBlank(host)){
+			logger.error("Host config parameter is missing");
+			setProperlyConfigured(false);
+			return;
+		}
+		
+		String portString = (String) configuration.get("port"); 
+		if(StringUtils.isNotBlank(portString))
+			port = Integer.parseInt(portString);
+
+		String prefix = "";
+		if(username != null) {
+			prefix = username + ":" + password + "@";
+		}
+		
+		baseURL = "http://" + prefix + host + ":" + port;
+
+		logger.debug("Autelius binding configured for host {}", host);
+		
+		setProperlyConfigured(true);
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "autelis";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		logger.trace("Connecting to {}" + baseURL);	
+
+		clearState();
+		
+		//we will reconstruct the document with all the responses combined for XPATH
+		StringBuilder sb = new StringBuilder("<response>");
+		
+		//pull down the three xml documents
+		String [] statuses = {"status", "chem", "pumps"};
+		for(String status : statuses){
+			String response = HttpUtil.executeUrl("GET", baseURL + "/" + status + ".xml", TIMEOUT);
+			logger.trace(baseURL + "/" + status + ".xml \n {}", response);
+			if(response == null){
+				logger.error("No response from Autelis controller!");
+				return;
+			}
+			//get the xml data between the response tags and append to our main doc
+			Matcher m = responsePattern.matcher(response);
+			if (m.find()) {
+			    sb.append(m.group(1));
+			}
+		}
+		//finish our "new" XML Document
+		sb.append("</response>");
+		
+		/*
+		 * This xmlDoc will now contain the three XML documents we retrieved wrapped in 
+		 * response tags for easier querying in XPath.
+		 */
+		String xmlDoc = sb.toString();
+		
+		for (AutelisBindingProvider provider : providers) {
+			for (String itemName : provider.getItemNames()) {
+				Item item = provider.getItem(itemName);
+				String config = provider.getAutelisBindingString(itemName);
+				XPathFactory xpathFactory = XPathFactory.newInstance();
+				XPath xpath = xpathFactory.newXPath();
+				try {
+					
+					InputSource is = new InputSource(new StringReader(xmlDoc));
+					String value = xpath.evaluate("response/" + config.replace('.', '/'), is);
+					State state = toState(item.getClass(), value);
+					State oldState = stateMap.put(itemName, state);
+					if(!state.equals(oldState)){
+						logger.debug("updating item {} with state {}", itemName, state);
+						eventPublisher.postUpdate(itemName, state);
+					}
+				} catch (XPathExpressionException e) {
+					logger.error("could not parse xml",e);
```

Comment:
decrease to WARN since the overall openHAB stability is not affected by this issue